throttler.go

```diff
@@ -44,13 +53,24 @@ func NewThrottler() Throttler {
 }
 
 // IsThrottled checks if the error has been throttled. If not, throttle it
-func (t *Throttler) IsThrottled(ocError error) bool {
+func (t *Throttler) IsThrottledOrGraced(ocError error) bool {
 	dc, err := t.getDiskCache()
 	if err != nil {
 		return false
 	}
-	cachedTime, throttled := dc.Get(ocError.Error())
 
+	cachedDetectionTime, hasCachedDetectionTime := dc.Get(fmt.Sprintf("%v_detectionTime", ocError.Error()))
```

Comment:
Not in the correct place here. Grace's cache key should be set along when Throttle's key is set.

Reply:
Thank you, let me try it

Reply:
Hi @chetra-tep,
I have optimized the function and tried different approaches for the **Cache key**. The key is set in the same function just before the **Throttle key**. So if I put the function into the **Throttle function**, then it will be more difficult to check the **Grace period** on the first iteration. 

I think there is not much difference if it's in another function since they are still in the same parent function `IsThrottledOrGraced`, and will be executed sequentially. Is there any scenario when the key shouldn't be handled at that place? 

Please let me know what you think or if I misunderstood anything

Reply:
I ll test it first.

Reply:
Doesn't work

```
	os.Setenv("APP_NAME", "123")
	os.Setenv("MS_TEAMS_ALERT_ENABLED", "test")
	os.Setenv("MS_TEAMS_WEBHOOK", "true")
	os.Setenv("THROTTLE_DURATION", "1")
	os.Setenv("THROTTLE_GRACE_SECONDS", "5")
	t := n.NewThrottler()
	t.CleanThrottlingCache()

	err := errors.New("ignore 000")
	alert := n.NewAlert(err, nil)
	alert.Notify() // false, caught by grace
	time.Sleep(6 * time.Second)

	alert.Notify() // true, first alert
	alert.Notify() // false, second alert, caught by throttle
        time.Sleep(61 * time.Second)
	alert.Notify() // false, caught by grace
	alert.Notify() // true
	alert.Notify() // false, caught by throttle
```

output

```
should alert: false
should alert: true
should alert: false
should alert: false // both 6 seconds & 1 min+1 sec have passed and still false
should alert: false
should alert: false
```

Reply:
Doesn't work

```
should alert: false
should alert: true
should alert: false
should alert: false
should alert: false
should alert: false
```

```
func main() {
	os.Setenv("APP_NAME", "123")
	os.Setenv("MS_TEAMS_ALERT_ENABLED", "test")
	os.Setenv("MS_TEAMS_WEBHOOK", "true")
	os.Setenv("THROTTLE_DURATION", "1")
	os.Setenv("THROTTLE_GRACE_SECONDS", "5")
	t := n.NewThrottler()
	t.CleanThrottlingCache()

	err := errors.New("ignore 000")
	alert := n.NewAlert(err, nil)
	alert.Notify()              // false, caught by grace
	time.Sleep(6 * time.Second) // wait for grace to go over

	alert.Notify() // true, first alert
	alert.Notify() // false, caught by throttle
	time.Sleep(61 * time.Second)
	alert.Notify()              // false, caught by grace
	time.Sleep(6 * time.Second) // wait for grace to go over
	alert.Notify()              // true
	alert.Notify()              // false, caught by throttle
}


```

Reply:
thank you, I've adjusted the grace reset, so it should work now

Reply:
Will test. Thanks.